### ISSUE TITLE
how to compute the mismatches / indels for the assembly

### ISSUE THREAD

**Issue by xiaoluo91** [ablab/quast#148]
Hi, I see in the quast document:

-mismatches per 100 kbp is the average number of mismatches per 100000 aligned bases. True SNPs and sequencing errors are not distinguished and are counted equally.

-indels per 100 kbp is the average number of indels per 100000 aligned bases. Several consecutive single nucleotide indels are counted as one indel.

What's does "per 100000 aligned bases" mean, is it for the assembled contigs or the reference genome? If this is for the reference genome, then the number of mismatches/indels which belong to overlapping regions between contigs will be accumulated, right ? Could you please explain this a little bit more ? Many thanks in advance!

**Comment by alexeigurevich** [comment]
Hi!
Thank you for the great question.
We tried to improve the documentation regarding this in the master branch and in our [v.5.1.0 release candidate](https://github.com/ablab/quast/releases/tag/quast_5.1.0rc1). I also wrote a detailed document ([v.5.1 release notes](https://docs.google.com/document/d/1aEs0lGRMCGWOyI_VCZw7AFyfK5tTxqUtwK7bpDC0aTs/edit?usp=sharing)) that includes a separate [section](https://docs.google.com/document/d/1aEs0lGRMCGWOyI_VCZw7AFyfK5tTxqUtwK7bpDC0aTs/edit#heading=h.wu3sb6bfqskn) about mismatch/indel calculation since it is a bit different in v.4.6, v.5.0 and v.5.1 (the current master branch). This is especially important if there are long overlapping fragments in the assembly (Duplication ratio >> 1.0). 

Briefly speaking: 
* in the master branch and in v.5.1.0rc1 it is counted as the number of mismatches/indel in the (aligned) assembly fragments normalized by the total (aligned) assembly length
* in v.5.0 it is counted as the number of mismatches/indel in the (aligned) assembly fragments normalized by the total length of the (assembled part of the) reference
* in v.4.6 and before it is counted as the number of mismatches/indel in the (assembled part of the) reference normalized by the total length of the (assembled part of the) reference

Change in v.5.0 was mostly caused by a switch from Nucmer to Minimap2 aligner and corresponding changes in the SNP computation logic. However, calculating the number of mismatches/indels in one thing (aligned assembly) and normalizing by another thing (assembled part of the reference) is awkward so we fixed this in v.5.1.
